tests/test_client_helpers.py

```diff
@@ -0,0 +1,107 @@
+    tag_name = f"Change_{uuid.uuid4()}"
+    client_helpers.create_tag(
+        client=fs.client, repository=repository, ref=temp_branch, tag=tag_name
+    )
+
+    assert any(
+        commit.get("id") == tag_name for commit in fs.client.tags_api.list_tags(repository).results
+    )
+
+
+def test_get_tags(fs: LakeFSFileSystem, repository: str) -> None:
+    response = client_helpers.get_tags(client=fs.client, repository=repository)
+    assert "pagination" in response
+    assert "results" in response
+
+
+def test_merge_into_branch(
+    random_file_factory: RandomFileFactory, fs: LakeFSFileSystem, repository: str, temp_branch: str
+) -> None:
+    random_file = random_file_factory.make()
+
+    new_branch = "new_branch_" + "".join(random.choices(string.digits, k=8))
+    source_ref = f"{repository}/{new_branch}/{random_file.name}"
+    with fs.scope(create_branch_ok=True):
+        fs.put(str(random_file), source_ref)
+        client_helpers.commit(
+            client=fs.client, repository=repository, branch=new_branch, message="Commit new file"
+        )
+
+    assert (
+        len(
+            fs.client.refs_api.diff_refs(
+                repository=repository, left_ref=temp_branch, right_ref=new_branch
+            ).results
+        )
+        > 0
+    )
+    client_helpers.merge(
+        client=fs.client, repository=repository, source_ref=new_branch, target_branch=temp_branch
+    )
+    assert (
+        len(
+            fs.client.refs_api.diff_refs(
+                repository=repository, left_ref=temp_branch, right_ref=new_branch
+            ).results
+        )
+        == 0
+    )
+
+
+def test_merge_into_branch_aborts_on_no_diff(
+    caplog: pytest.LogCaptureFixture, fs: LakeFSFileSystem, repository: str, temp_branch: str
+) -> None:
+    client_helpers.merge(
+        client=fs.client, repository=repository, source_ref="main", target_branch=temp_branch
+    )
+    assert caplog.records[0].message == "No difference between source and target. Aborting merge."
+
+
+def test_revert(
+    fs: LakeFSFileSystem, random_file_factory: RandomFileFactory, repository: str, temp_branch: str
+) -> None:
+    random_file = random_file_factory.make()
+    source_ref = f"{repository}/{temp_branch}/{random_file.name}"
+    with fs.scope(create_branch_ok=True):
+        fs.put(str(random_file), source_ref)
+        client_helpers.commit(
+            client=fs.client, repository=repository, branch=temp_branch, message="Commit new file"
+        )
+    assert (
+        len(
+            fs.client.refs_api.diff_refs(
+                repository=repository, left_ref="main", right_ref=temp_branch
+            ).results
+        )
+        > 0
+    )
+    client_helpers.revert(client=fs.client, repository=repository, branch=temp_branch)
+    assert (
+        len(
+            fs.client.refs_api.diff_refs(
+                repository=repository, left_ref="main", right_ref=temp_branch
+            ).results
+        )
+        == 0
+    )
```

Comment:
Technically OK, you think it might be easier to test on number of commits (from n to n+1 back to n)?

Reply:
AFAIK the commit revert is implemented as an additional commit. So testing for a reduced number of commits does not work. The increase in commits could then also be due to other changes on the branch. Thats why I think checking for the differences on the branch makes most sense here.

Reply:
Of course, that was my bad. But still, n->n+1->n+2 should work, no? We are on a temporary branch, and not in a race condition against anyone, so the number of commits should not increase.

(n would have to be set as the number of commits before the one made in the test.)